Adding API docs source files

sshniro opened the pull request.

thc202 commented:
Do we need `apis.md` checked in? Isn't that automatically generated by the build?

kingthorin commented on source/includes/attack.md:
````diff
@@ -0,0 +1,218 @@
+$ curl "http://localhost:8080/JSON/ascan/action/stop/?apikey=<ZAP_API_KEY>&scanId=<SCAN_ID>"
+```
+
+The [scan](#ascan_scan_api) endpoint runs the active scanner against the given URL and/or Context. Optionally, the 'recurse' parameter can be used to scan URLs 
````
Let's switch `and/or` to `or`.

Ref: http://kingthorin.blogspot.com/2007/09/writing-rant.html

kingthorin commented on source/includes/attack.md:
```diff
@@ -0,0 +1,218 @@
+Active scanning attempts to find potential vulnerabilities by using known attacks against the selected targets. Active scanning 
+is an attack on those targets. You should **NOT** use it on web applications that you do not own. 
+
+### Start Active Scanner
```
Headings in all content should be consistent, with or without leading space. (For my 2 cents I think with is better.)

sshniro replied:
Refactored all headings with leading space after #.

kingthorin commented on source/includes/contributing.md:
```diff
@@ -0,0 +1,182 @@
+**2. REST API Documentation**
+
+    Rest API is documented using the Open API specification. The specification could be improved by enhacing the description of 
+    parameters/ results/ data types etc. The open API specification is avaiable in [Github](https://github.com/zaproxy/zap-api-docs/blob/master/openapi.yaml).
```
> ... is available via

kingthorin commented on source/includes/contributing.md:
```diff
@@ -0,0 +1,182 @@
+
+**3. Feature Documentation**
+
+    Feature documentation of the ZAP is avaiable in the [Github wiki](https://github.com/zaproxy/zaproxy/wiki).
```
> Feature documentation for ZAP is available on the Github wiki

@psiinon @thc202 should this link to core-help and zap-extensions?

sshniro replied:
I'll link all the repositories here.

kingthorin commented on source/includes/contributing.md:
```diff
@@ -0,0 +1,182 @@
+**1. "Edit this File on Github" Option**
+    
+You can edit the documentation in the browser via navigating to the relevant source file and clicking the edit this file button.
+This workflow is recommended for minor changes, For example correcting typos/spellings/grammar etc.
```
After the comma shouldn't be a capital.

sshniro replied:
Resolved.

kingthorin commented on source/includes/contributing.md:
```diff
@@ -0,0 +1,182 @@
+
+**2. Local Setup and Editing**
+    
+You can fork the repository in Github and submit the changes via [pull requests](https://help.github.com/en/articles/creating-a-pull-request-from-a-fork). 
```
It seems like in vs on Github applies throughout.

kingthorin commented on source/includes/contributing.md:
```diff
@@ -0,0 +1,182 @@
+* Data types: `json`, `xml`, `html`
+* File name: `test.py`, `/path-to-your-data/xml/example-name`
+
+**Code block**                                              
```
Title caps?

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+The following are some of the options to explore the site by using ZAP. You can use multiple approaches in a combination to
+get more complete coverage of the application.
+
+* **Traditional Spider (Crawler):** Use this approach to crawl the HTML resources (hyperlinks etc) in the web application
```
Period

sshniro replied:
Resolved.

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+
+### Start the Spider
+
+The Spiders explore the site and they don't actually do any scanning. The [scan](#spider_scan_api) API runs the spider against the given URL. 
```
Do we need some sort of clarification here that while the Spiders themselves don't scan their traffic is passively scanned? (Or am I over thinking it?)

sshniro replied:
Added an extra line: The resources crawled by the Spider(s) are passively scanned in the background via the Passive Scanner.

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+The Spiders explore the site and they don't actually do any scanning. The [scan](#spider_scan_api) API runs the spider against the given URL. 
+Optionally, the 'maxChildren' parameter can be set to limit the number of children scanned and the 'recurse' parameter can 
+be used to prevent the spider from seeding recursively. The parameter 'subtreeOnly' allows to restrict the spider under a 
+site's subtree (using the specified 'url'). The parameter 'contextName' can be used to constrain the scan to 
```
Full caps for URL?

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+site's subtree (using the specified 'url'). The parameter 'contextName' can be used to constrain the scan to 
+a Context. View the [context example](#context_advanced) to understand how to create a context with ZAP API.   
+
+The code sample on the right recursively scans the web application with the provided URL. The scan ID is returns as a reponse
```
> The scan ID is returned

(Past tense vs plural)

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+### View Spider Results
+
+The results of the crawling can be obtained via the [results](#spider_results_api) API. The following image shows the JSON sample 
+response provided by the results API, enlisitng all the resources crawled by Spider.
```
enlisting or just "listing"?

sshniro replied:
changed to "listing"

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+
+### Stop or Pause the Spider
+
+If the scanning takes too much time than expected you can stop or pause the scanning via using the [stop](#spider_stop_api) 
```
> If the scanning takes more time than expected

??

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+            String status;
+
+            long startTime = System.currentTimeMillis();
+            long timeout = 120000; // Two minutes in milli seconds
```
Maybe this would be simpler for users as: `long timeout = TimeUnit.MINUTES.toMillis(2);`?

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+ 
+Use the Ajax Spider if you may have web applications written in Ajax. The Ajax Spider allows you to crawl web applications 
+written in Ajax in far more depth than the native Spider.You should also use the native Spider as well for complete coverage 
+of a web application (e.g. to cover HTML comments).
```
Instead of "written in Ajax" perhaps "heavily dependent on Ajax or JavaScript" ?

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+### Start Ajax Spider
+
+The scan API starts the Ajax Spider to given URL. Similar to the Traditional Spider, Ajax Spider can be also limited to a 
+context or scope. The parameter 'contextName' can be used to constrain the scan to a Context, the option 'in scope' is 
```
Is it really 'in scope' with a space? (elsewhere you've used the actual param name or seem to have)

kingthorin replied:
This is still outstanding

sshniro replied:
Changed to `inScope`

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+
+The scan API starts the Ajax Spider to given URL. Similar to the Traditional Spider, Ajax Spider can be also limited to a 
+context or scope. The parameter 'contextName' can be used to constrain the scan to a Context, the option 'in scope' is 
+ignored if a context was also specified. The parameter 'subtreeOnly' allows to restrict the spider under a site's subtree (using the specified 'url'). 
```
Full caps on URL?

sshniro replied:
Resolved

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+
+### Start Ajax Spider
+
+The scan API starts the Ajax Spider to given URL. Similar to the Traditional Spider, Ajax Spider can be also limited to a 
```
Instead of "to a given URL" perhaps "for" or "based on"?

sshniro replied:
Resolved

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+Unlike the traditional Spider, Ajax Spider does not provide a percentage for the work to be done. Use the [status](#spider_status_api) 
+endpoint to identify whether the Ajax Spider is still active or finished.
+
+
```
Remove a blank line?

sshniro replied:
Removed

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+
+### View Results
+
+Similar to the Traditional Spider, Ajax Spider's [results](#aspider_results_api) API can be used to view the resources 
```
> ... Traditional Spider, the Ajax Spider's ...

sshniro replied:
Resolved

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+
+### Stop the Ajax Spider
+
+Ajax spider does not have an indication on how much resources are left to be crawled. Therefor if the Ajax spider takes too much time
```
"Therefore" with a trailing 'e'? (I think I may ahve seen a missed this somewhere earlier too.)

sshniro replied:
Resolved

kingthorin commented on source/includes/explore.md:
```diff
@@ -0,0 +1,285 @@
+Ajax Spider.
+
+<aside class="success">
+Well done! Now ZAP has crawled the application using the Spider and the Ajax Spider. Move on to the attacking section to learn how 
```
Should we be slightly less specific/prescriptive here?

> using the Spider or the Ajax Spider.

Since user's aren't required to use both.

sshniro replied:
Resolved

kingthorin commented on source/includes/results.md:
```diff
@@ -0,0 +1,124 @@
+#Getting the Results
+
+
```
Remove a blank line?

kingthorin replied:
^

kingthorin commented on source/includes/results.md:
````diff
@@ -0,0 +1,124 @@
+print('Total number of alerts: ' + str(alert_count))
+```
+
+```java
````
Is the order of these example sections relevant? I think most of the earlier content was java first...

sshniro replied:
Changed the order in all files to python -> java -> shell

kingthorin commented on source/includes/results.md:
```diff
@@ -0,0 +1,124 @@
+$ curl "http://localhost:8080/JSON/alert/view/alertsSummary/?apikey=<ZAP_API_KEY>baseurl=<BASE_URL>"
+
+# To view alerts by risk category
+http://localhost:8080/JSON/alert/view/alertsByRisk/?apikey=<ZAP_API_KEY>&url=<BASE_URL>&recurse=
```
Is something missing in this example? (Ex: prompt/command/quotes)

kingthorin replied:
^

sshniro replied:
Curl statement added

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+web applications. ZAP also has an extremely powerful API that allows you to do nearly everything that is possible via the desktop interface.
+This allows the developers to automate pentesting and security regression testing of the application in the CI/CD pipeline. 
+
+
```
Remove one of these blank lines?

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+the right; switch the programming language of the examples with the tabs on the top right. 
+If anything is missing or seems incorrect, please check the [FAQs](https://github.com/zaproxy/zaproxy/wiki/FAQtoplevel) or the
+[GitHub issues](https://github.com/zaproxy/zaproxy/issues) for existing known issues.
+Also, if you are new to ZAP, then check out the [getting started guide](https://github.com/zaproxy/zaproxy/releases/download/2.4.0/ZAPGettingStartedGuide-2.4.pdf) 
```
This should use 2.8 links.

kingthorin replied:
^

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+* [**API Catalogue**](#api-catalogue) section contains Open API definitions and auto generated code for ZAP APIs. 
+
+<aside class="notice">
+The examples shows some usages with the minimal required arguments. However, this is not a reference, and not all APIs 
```
"shows" should be "show"

kingthorin replied:
^

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+
+<aside class="notice">
+The examples shows some usages with the minimal required arguments. However, this is not a reference, and not all APIs 
+nor arguments are shown. View the API catalog to see all the parameters and scope of each APIs.
```
Need to pick a consistent spelling for "catalog" or "catalogue" throughout.

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+nor arguments are shown. View the API catalog to see all the parameters and scope of each APIs.
+</aside>
+
+
```
Remove an extra blank line?

kingthorin replied:
^

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+![zap_desktop_api](../images/zap_desktop_api.png)
+
+<aside class="notice">
+ZAP requires API Key to perform specific actions via the REST API. The API key must be specified on all API 'actions' and some 'other' operations. 
```
> ZAP requires an API Key

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+### API URL Format
+
+The API is available via `GET` and `POST` endpoints and the response is available in `JSON`, `XML`, and `HTML` formats. All the 
+response formats returns the same information, just in a different format. Based on the use case, choose the appropriate format. 
```
"returns" should be "return"

kingthorin replied:
^

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+response formats returns the same information, just in a different format. Based on the use case, choose the appropriate format. 
+For example, to generate easily readable reports use the HTML format and use XML/JSON based response to parse the results quickly.
+
+The following shows the API URL format of ZAP:
```
> The following example shows...

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+
+##Quick Setup Guide
+
+
```
Remove a blank line?

kingthorin replied:
^

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+The quick setup guide focuses on setting up _ZAP_ and a testing application. If you have already setup ZAP then Jump to 
+specific [example](#exploring-the-app) to experiment with specific features.
+
+###Install  ZAP
```
This seems to be starting not installing....

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+// For Linux, Option: 2, using normal/ Desktop interface mode
+<ZAP_HOME>./zap.sh
+
+# For Windows, Run the exe file or zap.bat script to start the ZAP
```
Use `//`/ here instead of `#`?

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+# For Linux, Option: 2, using normal/ Desktop interface mode
+$ <ZAP_HOME>./zap.sh 
+
+# For Windows, Run the exe file or zap.bat script to start the ZAP
```
Not "the ZAP", just "ZAP"...a few occurrences below too...

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+The example guide uses [Google's Firing Range](https://github.com/google/firing-range) and 
+[OWASP Juice Shop](https://github.com/bkimminich/juice-shop) to perform the security testing. 
+The Spidering and Attacking examples use the [public instance](https://public-firing-range.appspot.com) of the 
+firing range, and OWASP Juice Shop is used to showcase the Authentication examples of ZAP. 
```
> Juice Shop are used ...

kingthorin commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,168 @@
+firing range, and OWASP Juice Shop is used to showcase the Authentication examples of ZAP. 
+
+The following is a [list](https://www.owasp.org/index.php/OWASP_Vulnerable_Web_Applications_Directory_Project#tab=On-Line_apps) 
+of publicly available vulnerable applications that you can also use in conjunction with ZAP.
```
> ... also be used in ...

sshniro commented:
> Do we need `apis.md` checked in? Isn't that automatically generated by the build?

@thc202 The apis.md is created by another [node module](https://github.com/Mermade/widdershins). As the build is triggered with Ruby and I have also changed some titles, therefore I did not include this in the build pipeline. I was planning to do it after finalizing the swagger files.

kingthorin commented:
There's a ton of stuff from my previous review still outstanding. (You need to un-hide the hidden items, there should be a banner/link.)

sshniro commented on source/includes/results.md:
```diff
@@ -0,0 +1,124 @@
+# To view the alerts
+$ curl "http://localhost:8080/JSON/alert/view/alerts/?apikey=<ZAP_API_KEY>&baseurl=<BASE_URL>&start=0&count=5000&riskId="
+
+# To view the summay of the alerts
```
Resolved.

sshniro commented on source/includes/explore.md:
````diff
@@ -0,0 +1,286 @@
+```
+ 
+Use the Ajax Spider if you may have web applications written in Ajax. The Ajax Spider allows you to crawl web applications 
+written in Ajax in far more depth than the native Spider.You should also use the native Spider as well for complete coverage 
````
Refactored to traditional.

sshniro commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,167 @@
+
+<aside class="notice">
+The examples shows some usages with the minimal required arguments. However, this is not a reference, and not all APIs 
+nor arguments are shown. View the API catalog to see all the parameters and scope of each APIs.
```
Refactored to `the parameters and scope of each API.`

sshniro commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,167 @@
+
+### Client SDKs
+
+ZAP provides official clients for Python, Java, and Node JS. Visit the following link to download the [official SDKs](https://github.com/zaproxy/zaproxy/wiki/ApiDetails). 
```
`ZAP provides official clients for many languages such as Python, Java, NodeJS, and .Net. Visit the following link to view and download all supported [SDKs](https://github.com/zaproxy/zaproxy/wiki/ApiDetails). `

sshniro commented on source/includes/welcome.md:
````diff
@@ -0,0 +1,167 @@
+``` shell
+# For Linux/, Option: 1, using "headless/daemon" mode
+$ <ZAP_HOME>./zap.sh -daemon -config api.key=change-me-9203935709
+# For Linux, Option: 2, using normal/ Desktop interface mode
````
`// For Linux, Option: 2, using ZAP desktop App`

sshniro commented on source/includes/welcome.md:
Refactored all occurrences

sshniro commented on README.md:
```diff
@@ -0,0 +1,17 @@
+This allows the developers to automate pentesting and security regression testing of the application in the CI/CD pipeline.
+This repository This provides example guides & API definitions for ZAP APIs.
+
+The live demo can be viewed in the following [URL](https://zaproxy.github.io/zap-api-docs).
```
@thc202  Shall we change this when we get a final link? Will this be zaproxy.org/docs or docs.zaproxy.org ?

thc202 replied:
We can change later, the final path is still not decided (AFAIK).

sshniro commented on source/includes/attack.md:
````diff
@@ -0,0 +1,218 @@
+```
+
+All requests that are proxied through ZAP or initialised by tools like the Spider are passively scanned. You do not have 
+to manually start the passive scan process, ZAP by default passively scans all HTTP messages (requests and responses) sent 
````
`ZAP by default passively scans all HTTP and WebSocket messages (requests and responses)  
which are sent to the application. `

sshniro commented:
@thc202 @kingthorin Thank you for the comments, I resolved the requested changes. Shall we merge this PR if no additional review is pending?

sshniro commented:
> * For consistency we should use the same term/word when referring to the target site, it's using `App`, `application`, `web application`, `site`.
> * Do we need `source/examples/jsFiddle/`?

I have refactored `web application` to `application` and removed the jsFiddle content.

thc202 commented on source/includes/explore.md:
```diff
@@ -0,0 +1,286 @@
+Ajax spider does not have an indication on how much resources are left to be crawled. Therefor if the Ajax spider takes too much time
+than expected, then it can be stopped by using the [stop](#aspider_stop_api) API.
+
+View the [advanced section on Ajax Spider](#ajax-spider-settings) section to learn more about how to further fine-tune or improve the results of the 
```
Was referring to "View the advanced section on Ajax Spider **section** to", it seems redundant(?).

sshniro replied:
aw, got it wrong. Now it's fixed.

thc202 commented on source/includes/explore.md:
````diff
@@ -0,0 +1,284 @@
+$ curl "http://localhost:8080/JSON/AjaxSpider/action/stop/?apikey=<ZAP_API_KEY>"
+```
+ 
+Use the Ajax Spider if you have applications which heavily dependent on Ajax or JavaScript. The Ajax Spider allows you to crawl web applications 
````
`dependent` → `depend`?

thc202 commented on source/includes/contributing.md:
```diff
@@ -0,0 +1,183 @@
+## Markdown Syntax
+
+The API docs are created using standard markdown files. This section contains information regarding the syntax and linting of the Markdown files.
+Refer to the [Slate documentation](https://github.com/slatedocs/slate/wiki/Markdown-Syntax). Also Refer to this 
```
`Also Refer to` the "refer" should be lower case.

thc202 commented on source/includes/welcome.md:
```diff
@@ -0,0 +1,164 @@
+* [**Getting the Results**](#getting-the-results) section contains examples on how to retrieve alerts and generate Reports from ZAP.
+* [**Getting Authenticated**](#getting-authenticated) section contains examples on how to authenticate the web application with ZAP.
+* [**Advanced Settings**](#advanced-settings) section contains advanced configurations on how to fine tune ZAP results.
+* [**Contributions**](#contributions-welcome) section contains guidelines and instructions on how to contribute to ZAP's documentations.
```
`documentations` → `documentation`

kingthorin commented:
Looks like we're down to three minor changes.

thc202 commented:
Thanks!
